src/VTK/VtkGrid2DRepresentation.cxx

```diff
@@ -38,10 +34,8 @@ VtkGrid2DRepresentation::~VtkGrid2DRepresentation()
 		epcPackageSubRepresentation = nullptr;
 	}
 
-	if (grid2DPoints != nullptr) {
-		delete grid2DPoints;
-		grid2DPoints = nullptr;
-	}
+
+	grid2DPoints.~VtkGrid2DRepresentationPoints();

```

Comment:
No need at all. Only a destructor on a pointer is needed to be called.
This variable here is automatically deleted automatically (that's one of the advantage compared to using pointer)